src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1329,6 +1310,10 @@ export class SettingTextRenderer extends AbstractSettingRenderer implements ITre
 	}
 
 	protected renderValue(dataElement: SettingsTreeSettingElement, template: ISettingTextItemTemplate, onChange: (value: string) => void): void {
+		if (!template.inputBox) {
+			template.inputBox = this.createInputBox(dataElement, template);
```

Comment:
Templates get reused for different setting elements. Here you might create one for a multiline element then reuse it for a non-multiline element. I'd recommend having a different template type and renderer for multiline strings, instead of recreating the control